### ISSUE TITLE
High Priority A - Keyboard navigation for drop-down elements

### ISSUE THREAD

**Issue by gingerbenw** [datasciencecampus/sdg-indicators#1065]
The drop-down and accordion elements aren’t accessible to users of assistive technologies as non-interactive elements have been used and customised.  Keyboard only users would expect to be able to navigate to the drop-down box using the tab key and open it with the spacebar or enter key.   Currently, users that navigate with assistive technologies are not able to access the filter options within the drop-box or the checkboxes within them. This also applied to the accordion type elements. 

**Note:** This issue is present throughout the templates where the accordion and drop-down features are. 

**Code Ref(s):**    
```html
<div class="variable-selector" data-field="Group">
  <h5>Group<i class="fa fa-chevron-down"></i></h5>
  <div class="bar">
    <div class="selected" style="width: 100%;"></div>
  </div>
  <div class="variable-options" style="display: none;">
    <div>
      <button data-type="select">Select all</button>
      <button data-type="clear">Clear all</button>
    </div>
    <label class="default"><input type="checkbox" value="A" data-field="Group">A</label>
    <label class="default"><input type="checkbox" value="B" data-field="Group">B</label>        
  </div>     
</div>
```

**Solution:** We strongly recommend using standard HTML elements to ensure that roles, states and controls and exposed correctly, enabling all users regardless of assistive technology to access and interact with them. The main considerations when implementing custom elements: 
1. Ensure that the element is focusable and that users are able to gain access via the keyboard and determine this clearly 
2. Ensure that the element is operable via keyboard alone 
3. Ensure that the element operates as expected using the standard keys   
4. Ensure a clear indication of focus  
5. Ensure that the element is labelled and is exposed as an accessible name in accessibility APIs 
6. Ensure that the role is exposed in accessibility APIs i.e. role= “button” 
7. Ensure that all states and properties are clearly defined and exposed in accessibility APIs 
8. Ensure that in high contrast mode the control is visible and usable in High Contrast Mode (colour inversion)

**Comment by gingerbenw** [comment]
the keyboard navigation can be solved by adding a `tabindex=''` to each expandable item within the template.

**Comment by AnnCorp** [comment]
From Deb Roberts at DAC:
There is a link below that gives an accessible dropdown example, that is tried and tested by our team and found to be accessible.  The ‘Billing Address’ drop down is the one that you require.
https://www.w3.org/TR/wai-aria-practices/examples/accordion/accordion.html

**Comment by gingerbenw** [comment]
found the missing component! simply had to add a button tag around the item we wanted to trigger 👍

**Comment by gingerbenw** [comment]
#1311

**Comment by dougmet** [comment]
yes go for it. Thanks @gingerbenw
